source/docs/software/advanced-controls/introduction/tuning-elevator.rst

```diff
@@ -0,0 +1,137 @@
+
+.. raw:: html
+
+    <div class="viz-div" id="elevator_feedback_container">
+      <div >
+         <div class="col" id="elevator_feedback_plotVals"></div>
+         <div class="col" id="elevator_feedback_plotVolts"></div>
+      </div>
+      <div class="flex-grid">
+         <div class="col" id="elevator_feedback_viz"></div>
+         <div id="elevator_feedback_ctrls"></div>
+      </div>
+      <script>
+         arm_pidf = new VerticalElevatorPIDF("elevator_feedback", "feedback");
+      </script>
+    </div>
+
+Perform the following:
+
+1. Set :math:`K_p`, :math:`K_i`, and :math:`K_d` to zero.
+2. Increase :math:`K_p` until the mechanism responds to a sudden change in setpoint by moving sharply to the new position.  If the controller oscillates too much around the setpoint, reduce `K_p` until it stops.
+3. Increase :math:`K_i` when the :term:`output` gets "stuck" before converging to the :term:`setpoint`.
+4. Increase :math:`K_d` to help the system track smoothly-moving setpoints and further reduce oscillation.
+
+.. note:: Feedback-only control is not a good control scheme for vertical elevators!  Do not be surprised if/when the simulation below does not behave consistently, even when the "correct" constants are used.
+
+.. collapse:: Tuning solution
+
+   There is no perfect tuning solution for this control strategy.  Values of :math:`K_p = 10.0`,  :math:`K_i = 2.5` and and :math:`K_d = 0.0` yield a possible solution, but with overshoot and large settling times. Additionally, it will act very differently depending on the setpoint - aggressively overshooting at the top and undershooting at the bottom.
+
+
+Motion Profiled, Feedforward, and Feedback Control
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Interact with the simulation below to examine how the elevator system responds when controlled only by a feedforward controller.
+
+.. note:: To change the elevator setpoint, click on the desired height along the vertical support.
+
+.. raw:: html
+
+    <div class="viz-div" id="elevator_feedforward_container">
+      <div >
+         <div class="col" id="elevator_feedforward_plotVals"></div>
+         <div class="col" id="elevator_feedforward_plotVolts"></div>
+      </div>
+      <div class="flex-grid">
+         <div class="col" id="elevator_feedforward_viz"></div>
+         <div id="elevator_feedforward_ctrls"></div>
+      </div>
+      <script>
+         arm_pidf = new VerticalElevatorPIDF("elevator_feedforward", "both");
+      </script>
+    </div>
+
+
+To tune the feedforward controller, perform the following:
+
+1. Start with fairly slow maximum velocity and maximum acceleration. 0.3 for both is a good guess.
+2. Set :math:`K_g`, :math:`K_v`, :math:`K_a`, :math:`K_p`, :math:`K_i`, and :math:`K_d` to zero.
+3. Increase :math:`K_g` as much as you can without the elevator moving upward. You will have to zero in on :math:`K_g` fairly precisely (at least four decimal places).
```

Comment:
We should explain the discrepancy between the advice to go to 4 decimals and the ideal answer that is just 2.28